app/controllers/application_controller.rb

```diff
@@ -661,7 +662,11 @@ def authorize_mini_profiler
   def check_xhr
     # bypass xhr check on PUT / POST / DELETE provided api key is there, otherwise calling api is annoying
     return if !request.get? && (is_api? || is_user_api?)
-    raise ApplicationController::RenderEmpty.new unless ((request.format && request.format.json?) || request.xhr?)
+    unless ((request.format && request.format.json?) || request.xhr?)
+      # Tests usually aren't trying to request HTML pages. Give a special error
+      raise ApplicationController::CheckXhrForTest.new if Rails.env.test?
```

Comment:
Instead of doing this check at runtime, we can defined the error as a constant so that the check is only done.

Reply:
Do you mean switching out the method body based on Rails.env.test?

Reply:
I was thinking something like

```
ERROR_CLASS = Rails.env.test? ? ApplicationController::CheckXhrForTest : ApplicationController::RenderEmpty

def check_xhr
    ....
    raise ERROR_CLASS.new
end
```

Reply:
I think the cost of the check is so cheap and it only happens anyway in exceptional cases (which is key). Code is somewhat simpler this way, I say we just keep it.